src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,428 @@ class Guides::Database::Querying < GuideAction
+    query.age(30)
 
-    * `created_at` (automatically added)
-    * `updated_at` (automatically added)
-    * `id` (automatically added)
-    * `age`
-    * `name`
+    # The query will run once `each` is called
+    # Results are not cached so a request will be made every time you call `each`
+    query.each do |user|
+      pp user.name
+    end
+    ```
+
+    ## Simple Selects
+
+    > When doing a `SELECT`, Avram will select all of the columns individually (i.e. `users.id,
+    > users.created_at, users.updated_at, users.name, users.age, users.admin`) as opposed to `*`.
+    > However, for brevity, we will use `COLUMNS`.
 
+    ### Select all
 
-    ## Distinct on
+    `SELECT COLUMNS FROM users`
 
     ```crystal
-    UserQuery.new.distinct_on(&.name)
+    users = UserQuery.new.all
     ```
 
-    ## Where clauses
+    ### Select first
 
-    You can use the methods generated by Lucky to create where clauses.
+    `SELECT COLUMNS FROM users LIMIT 1`
 
     ```crystal
-    # SELECT * FROM users where name='Paul';
-    UserQuery.new.name("Paul")
+    # raise Avram::RecordNotFound if nil
+    user = UserQuery.new.first
 
-    # SELECT * FROM users where age=28;
-    UserQuery.new.age(28)
+    # returns nil if not found
+    user = UserQuery.new.first?
     ```
 
-    ### Parsing query params
+    ### Select last
 
-    Lucky will accept the type defined in the model (e.g. `Int32` for `age`) or a
-    `String`. If you pass it a `String` then Lucky will attempt to parse it and use
-    the type defined for the column.
+    `SELECT COLUMNS FROM users ORDER BY users.id DESC LIMIT 1`
 
-    For example:
+    ```crystal
+    # raise Avram::RecordNotFound if nil
+    user = UserQuery.new.last
+
+    # returns nil if not found
+    user = UserQuery.new.last?
+    ```
+
+    ### Select by primary key
+
+    Selecting the user with `id = 3`.
+    `SELECT COLUMNS FROM users WHERE users.id = 3 LIMIT 1`
 
     ```crystal
-    # Lucky parses the string and use an Int32 in the query
-    UserQuery.new.age("30")
+    # raise Avram::RecordNotFound if nil
+    user = UserQuery.new.find(3)
+    ```
+
+    ### Select distinct / distinct on
 
-    # Won't accept other types
-    # Fails at compile time because you can't pass a Float64
-    UserQuery.new.age(30.0)
+    `SELECT DISTINCT COLUMNS FROM users`
+
+    ```crystal
+    UserQuery.new.distinct
     ```
 
-    > Note that this will change in the future but this is the current behavior. In
-    the future only the defined type (`Int32` in this example) will work. You will
-    then need to manually parse it. Don't worry though, Lucky will add some helpers
-    to actions and the model to make this easy and maintainable.
+    Select distinct rows based on the `name` column `SELECT DISTINCT ON (users.name) FROM users`
 
-    ## Ordering results
+    ```crystal
+    UserQuery.new.distinct_on(&.name)
+    ```
 
-    ````crystal
-    UserQuery.new.age.desc_order
-    # or
-    UserQuery.new.age.asc_order
```

Comment:
This is the type sage way of ordering. This should be the main way to order i  the “ordering” section. The order_by method is the non type safe but more flexible way since you can do string interpolation and stuff. It should be avoided if possible though